Src/BootCamp.Chapter/Program.cs

```diff
@@ -6,7 +6,84 @@ class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+
+            //Set variable
+            bool backtotop = true;
+            string option;
+
+            //Start While loop
+            while (backtotop == true)
+
+            {
+                backtotop = true;
+
+                //USer data input
+
+                Console.WriteLine("What is your first name?");
+                var firstname = Console.ReadLine();
+
+                Console.WriteLine("What is your second name?");
+                var secondname = Console.ReadLine();
+
+                Console.WriteLine("What is your age?");
+                var age = Console.ReadLine();
+
+                Console.WriteLine("What is your weight in kg?");
+                var weight = Console.ReadLine();
+
+                Console.WriteLine("What is your height in Meters?");
+                var height = Console.ReadLine();
+
+
+                //Convert vars
+                var weightsingle = Convert.ToSingle(weight);
+                var heightsingle = Convert.ToSingle(height);
+
+
+                //Calc for BMI
+                var Calc1 = weightsingle / heightsingle;
+                var BMI = Calc1 / heightsingle;
+
+
+                //Display output to user
+                Console.WriteLine(firstname + " " + secondname + "is " + age + " years old, his weight is " + weightsingle + " kg and his height is " + heightsingle + " cm ");
+                Console.WriteLine("Your BMI is " + BMI);
+
+
+                // Calc BMI scale 
+                if (BMI <= 18.5f)
+                {
+                    Console.WriteLine("You are underweight");
+                }
+                else if (BMI >= 18.6f && BMI <= 24.99f)
+                {
+                    Console.WriteLine("You are normal Weight");
+                }
+                else if (BMI >= 25 && BMI <= 29.99f)
+                {
+                    Console.WriteLine("You are overweight");
+                }
+                else if (BMI >= 30)
+                {
+                    Console.WriteLine("You are obese");
+                }
+
+                //Continue
+                Console.WriteLine("Would you like to do another person Y/N?");
+                option = Console.ReadLine();
+                if ((option) == "N" || (option) == "n")
+                {
+                    backtotop = false;
```

Comment:
A better way of doing this while loop is by using the do-while loop:

```
do 
{
    // all your code
    Console.WriteLine("Would you like to do another person Y/N?");
}while(string.Equals(!Console.ReadLine(), "N", StringComparison.OrdinalIgnoreCase));
```
This will always execute **at least once** and then after the first one it prompts the user if it wants to go again. Note the condition in the while loop parentheses. It's checking if the answer is no.

Reply:
Thank you for the feedback. I will take note and I will use this in future code.

Reply:
Not sure why @Alumniminium down thumbed this comment, but I noticed that I missed the actual reason to go for a do-while loop here over a while loop. The reason is that you're prompting the user to run again or not. Because this should not be prompted the first time the while loop runs, you need a boolean outside the while loop to facilitate this behavior. However, this is the exact equivalent of the do-while loop and the do-while loop here is much more intuitive and a better option because it indicates that the loop will always be executed once and after that, you can just prompt the user to run again. The answer will then be evaluated at the end of the loop, in the while condition.